tests/scenario/test_ingress_per_app.py

```diff
@@ -0,0 +1,172 @@
+# GIVEN a charm with ingress impl'd
+# WHEN a relation with traefik is formed
+# THEN traefik's config file's `server` section has all the units listed
+# AND WHEN the charm rescales
+# THEN the traefik config file is updated
+
+import tempfile
+from pathlib import Path
+
+import pytest
+import yaml
+from ops import pebble
+from scenario import Container, Context, Model, Mount, Relation, State
+
+
+@pytest.fixture
+def context(traefik_charm):
+    return Context(charm_type=traefik_charm)
+
+
+@pytest.fixture
+def model():
+    return Model(name="test-model")
+
+
+@pytest.fixture
+def temp_opt():
+    with tempfile.TemporaryDirectory() as td:
+        yield Path(td)
+
+
+@pytest.fixture
+def traefik_container(temp_opt):
+    layer = pebble.Layer(
+        {
+            "summary": "Traefik layer",
+            "description": "Pebble config layer for Traefik",
+            "services": {
+                "traefik": {
+                    "override": "replace",
+                    "summary": "Traefik",
+                    "command": '/bin/sh -c "/usr/bin/traefik | tee /var/log/traefik.log"',
+                    "startup": "enabled",
+                },
+            },
+        }
+    )
+
+    opt = Mount("/opt/", temp_opt)
+
+    return Container(
+        name="traefik",
+        can_connect=True,
+        layers={"traefik": layer},
+        service_status={"traefik": pebble.ServiceStatus.ACTIVE},
+        mounts={"opt": opt},
+    )
+
+
+@pytest.mark.parametrize("port, host", ((80, "1.1.1.1"), (81, "10.1.10.1")))
+@pytest.mark.parametrize("event_name", ("joined", "changed"))
+def test_ingress_per_app_created(
+    context, port, host, model, traefik_container, event_name, temp_opt
+):
+    """Check the config when a new ingress per leader is created or changes (single remote unit)."""
+    ipl = Relation(
```

Comment:
Typo?
```diff
-ipl
+ipa
```

Reply:
yes, thanks. I got so confused by that renaming thing...